### PR TITLE
VELOC updated to v.1.6

### PR DESCRIPTION
**Pull request by bnicolae**
Clean-up of dependencies

### CODE REVIEW

**Review by wspear — DISMISSED** [review]
Looks good to me. (Or, it will after fixing the flake8 style issue)

**Review by wspear — DISMISSED** [review]
Still looks good!

File: var/spack/repos/builtin/packages/veloc/package.py
Comment (becker33) [inline var/spack/repos/builtin/packages/veloc/package.py]: Isn't this version dependent? I don't think VeloC 1.0 requires c++17

File: var/spack/repos/builtin/packages/veloc/package.py
Comment (becker33) [inline var/spack/repos/builtin/packages/veloc/package.py]: If this is a new argument, it needs a version guard.

File: var/spack/repos/builtin/packages/veloc/package.py
Comment (becker33) [inline var/spack/repos/builtin/packages/veloc/package.py:27]: Old versions depend on boost. Try `depends_on('boost', when='@:1.5')`, or equivalent if I'm wrong about the version it changed in.

File: var/spack/repos/builtin/packages/veloc/package.py
Comment (becker33) [inline var/spack/repos/builtin/packages/veloc/package.py:49]: Boost argument is needed for older versions, right?

**Review by becker33 — CHANGES_REQUESTED** [review]

**Comment by bnicolae** [comment]
I think we are over-complicating things here. Probably it's for the best to simply remove the older versions, which nobody depends on anyway. The problem is that other libraries have evolved too in terms of API (e.g. AXL), so it is difficult to keep track of what version of VELOC needs what version of the dependencies. @becker33, would that be fine with you? I am of course happy to learn about how Spack normally handles this.

**Review by wspear — DISMISSED** [review]
This works for me.

**Comment by becker33** [comment]
@bnicolae Spack tries to maintain the ability to install older versions because people rely on them to reproduce older results. It's not enough to tell them to go back to an older version of Spack for the older versions, because newer versions of Spack may be necessary to run on newer systems.

The `when` clauses in Spack directives are designed to make it easy to support different requirements at different versions. Definitely let me know if there's some additional Spack feature you need to be able to express the veloc constraints, but I don't see any reason it shouldn't be easily expressible with the current features.

**Comment by eugeneswalker** [comment]
I added the version guards to address each of the points raised by @becker33:
https://github.com/ECP-VeloC/spack/pull/1

**Comment by bnicolae** [comment]
The APIs of the other dependencies (e.g. AXL) have changed from one version to another too. We didn't keep track of the list of dependency versions for each VELOC version. So, this would require and effort to test all combinations and write a rather complex Spack recipe to account for the past omissions. We do not have the manpower for such an effort. Also, none of our users has asked for historic support. Hence, my recommendation to simplify things and apply this best practice in the future.

**Comment by eugeneswalker** [comment]
> The APIs of the other dependencies (e.g. AXL) have changed from one version to another too. We didn't keep track of the list of dependency versions for each VELOC version. So, this would require and effort to test all combinations and write a rather complex Spack recipe to account for the past omissions. We do not have the manpower for such an effort. Also, none of our users has asked for historic support. Hence, my recommendation to simplify things and apply this best practice in the future.

If we use the version guards here: https://github.com/ECP-VeloC/spack/pull/1

At least for now that maintains the current behavior while allowing for the new version. We need to finalize all package updates tonight in order to begin final preparation for E4S 22.02 release, and we would like to include the new version of Veloc. Perhaps we could add the version guards and worry about the rest later? This at least preserves current behavior while allowing the new version in.

**Comment by bnicolae** [comment]
Ok, sounds good, I will pull your PR and adjust if needed.

**Comment by bnicolae** [comment]
To simply things, I used Boost for all versions. I've added a conditional dependency for AXL. I kept the conditional compiler flag.

**Comment by eugeneswalker** [comment]
I don't know why the linux test is failing the way it is, but there is a style issue with this PR.

If you run `spack style` locally inside your spack directory:
```
==> Running flake8 checks
  var/spack/repos/builtin/packages/veloc/package.py:38: [E303] too many blank lines (2)
  var/spack/repos/builtin/packages/veloc/package.py:52: [E999] SyntaxError: invalid syntax
  flake8 found errors
==> Error: spack style found errors
```

File: var/spack/repos/builtin/packages/veloc/package.py
Comment (bnicolae) [reply var/spack/repos/builtin/packages/veloc/package.py]: This is fixed now, cxx_17 applied only to v.1.6

File: var/spack/repos/builtin/packages/veloc/package.py
Comment (bnicolae) [reply var/spack/repos/builtin/packages/veloc/package.py]: Reverted to boost for all versions to simplify recipe

File: var/spack/repos/builtin/packages/veloc/package.py
Comment (bnicolae) [reply var/spack/repos/builtin/packages/veloc/package.py:27]: Now every version depends on boost

**Comment by alalazo** [comment]
Is this superseded by #34706 ? If not, can you rebase and ping me for a review once tests are passing?

**Comment by wspear** [comment]
> Is this superseded by #34706 ? If not, can you rebase and ping me for a review once tests are passing?

#34706 does (I thought) the bare minimum to get @1.6: to build and retains the old behavior for @:1.5. Though I note there are some dependencies I was told to add for 1.6 that don't seem to be used in this PR.

Deprecating older versions, updating the cmake version, adding bnicolae as a maintainer, updating the cflags and adding the boost dependency are all unique to this PR and I assume those changes are worth while.

**Comment by alalazo** [comment]
Agreed @wspear Ping me when ready.

**Comment by bnicolae** [comment]
All, FYI we will release VELOC 1.7 soon

**Comment by eugeneswalker** [comment]
Do we need any of these changes still?

If we can merge it soon, we will be able to include in E4S 23.08. But it has to be finished soon to give us time.

I submitted a PR here that adds v1.7:
* https://github.com/spack/spack/pull/39465

**Comment by bnicolae** [comment]
Thanks @eugeneswalker! Yes, all changes in this PR are needed in addition to the v1.7 PR. It all looks good to me.

**Comment by eugeneswalker** [comment]
I've consolidated all the changes in this PR over to:
* https://github.com/spack/spack/pull/39465

**Comment by eugeneswalker** [comment]
Can you leave a comment on the updated PR if you approve? I had to fix style issues and re-base on recent develop to pick up important changes to CI that will test veloc build before we merge:
* https://github.com/spack/spack/pull/39465

**Pull request closed without merging**
